internal/runners/packages/searchView.go

```diff
@@ -0,0 +1,304 @@
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/internal/logging"
+	"github.com/ActiveState/cli/internal/output"
+	"github.com/charmbracelet/bubbles/viewport"
+	tea "github.com/charmbracelet/bubbletea"
+	"github.com/charmbracelet/lipgloss"
+	"golang.org/x/term"
+)
+
+var (
+	keyName        = locale.Tl("search_name", "Name")
+	keyDescription = locale.Tl("search_description", "Description")
+	keyWebsite     = locale.Tl("search_website", "Website")
+	keyLicense     = locale.Tl("search_license", "License")
+	keyVersions    = locale.Tl("search_versions", "Versions")
+	keyVulns       = locale.Tl("search_vulnerabilities", "Vulnerabilities (CVEs)")
+
+	keys = []string{
+		keyName,
+		keyDescription,
+		keyWebsite,
+		keyLicense,
+		keyVersions,
+		keyVulns,
+	}
+)
+
+const (
+	leftPad        = 2
+	verticalMargin = 7
+	scrollUp       = "up"
+	scrollDown     = "down"
+)
+
+type errMsg error
+
+type view struct {
+	width         int
+	height        int
+	remaining     int
+	content       string
+	index         map[string]int
+	searchResults *structuredSearchResults
+	ready         bool
+	err           error
+	viewport      viewport.Model
+}
+
+func NewView(results *structuredSearchResults, out output.Outputer) (*view, error) {
+	outFD, ok := out.Config().OutWriterFD()
+	if !ok {
+		logging.Error("Could not get output writer file descriptor, falling back to stdout")
+		outFD = os.Stdout.Fd()
+	}
+
+	width, height, err := term.GetSize(int(outFD))
+	if err != nil {
+		return nil, errs.Wrap(err, "Could not get terminal size")
+	}
+
+	return &view{
+		width:         width,
+		height:        height,
+		searchResults: results,
+	}, nil
+}
+
+func (v *view) Init() tea.Cmd {
+	return nil
+}
+
+func (v *view) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
+	var (
+		cmd  tea.Cmd
+		cmds []tea.Cmd
+	)
+
+	switch msg := msg.(type) {
+	case tea.MouseMsg:
+		switch msg.Button {
+		case tea.MouseButtonWheelUp:
+			v.scroll(scrollUp, 3)
+			return v, nil
+		case tea.MouseButtonWheelDown:
+			v.scroll(scrollDown, 3)
+			return v, nil
+		}
+	case tea.KeyMsg:
+		stringMsg := strings.ToLower(msg.String())
+		switch stringMsg {
+		case "q", "ctrl+c":
+			return v, tea.Quit
+		case "up":
+			v.scroll(scrollUp, 1)
+			return v, nil
+		case "down":
+			v.scroll(scrollDown, 1)
+			return v, nil
+		case "pgup":
+			v.scroll(scrollUp, v.height-verticalMargin)
+			return v, nil
+		case "pgdown":
+			v.scroll(scrollDown, v.height-verticalMargin)
+			return v, nil
+		}
+	case tea.WindowSizeMsg:
+		if !v.ready {
+			// Keep the searching message and command in view
+			v.viewport = viewport.New(msg.Width, msg.Height-verticalMargin)
+			v.content = v.processContent()
+			v.viewport.SetContent(v.processContent())
+			v.initialRemaining()
+			v.ready = true
+		} else {
+			v.width = msg.Width
+			v.height = msg.Height
+		}
+	case errMsg:
+		v.err = msg
+		return v, nil
```

Comment:
How might this happen?

Reply:
Ahh good catch. This is a leftover from my initial attempts to get the view working and is no longer needed.